kodev

```diff
@@ -173,16 +174,31 @@ usage: build <OPTIONS> <TARGET>
 
 OPTIONS:
 
-    -v, --verbose   Build in verbose mode.
-    --debug         include debugging symbols (default for emulator)
-    --no-debug      no debugging symbols (default for target devices)
+    -v, --verbose   make the buildsystem more verbose
+    -d, --debug     include debugging symbols (default for emulator)
+    -n, --no-debug  no debugging symbols (default for target devices)
 
 TARGET:
 ${SUPPORTED_TARGETS}"
 
-    while [[ "${1}" == '-'* ]]; do
-        PARAM=$(echo "${1}" | awk -F= '{print $1}')
-        VALUE=$(echo "${1}" | awk -F= '{print $2}')
+    declare opt
+    declare -r E_OPTERR=85
+    declare -r short_opts="vhnd"
+    declare -r long_opts="verbose,help,no-debug,debug"
+
+    opt=$(getopt -o "${short_opts}" --long "${long_opts}" --name "kodev" -- "${@}")
+
+    # shellcheck disable=SC2181
+    if [[ $? -ne 0 ]]; then
```

Comment:
Keep in mind I'm just skimming, but why not listen to shellcheck? Seems easier to read to me. (Also you seem to be unnecessarily using Bash `[[ ]]`? :-P)

```suggestion
    

    # shellcheck disable=SC2181
    if ! opt=$(getopt -o "${short_opts}" --long "${long_opts}" --name "kodev" -- "${@}"); then
```

Reply:
I tend to default to `[[` ^^.

(And the one time I didn't was the one time I actually needed it, for the `gdb*` glob ;p

----

Yeah, the original check included a `$#` check which we don't need/handle differently, which is why it was split off. I'll tweak that ;).